packages/babel-cli/src/babel/file.js

```diff
@@ -6,10 +8,19 @@ import path from "path";
 import fs from "fs";
 
 import * as util from "./util";
-
-export default async function({ cliOptions, babelOptions }) {
-  function buildResult(fileResults) {
-    const map = new sourceMap.SourceMapGenerator({
+import { type CmdOptions } from "./options";
+
+type CompilationOutput = {
+  code: string,
+  map: Object,
+};
+
+export default async function({
+  cliOptions,
+  babelOptions,
+}: CmdOptions): Promise<void> {
+  function buildResult(fileResults: Array<Object>): CompilationOutput {
+    const map: Object = new sourceMap.SourceMapGenerator({
```

Comment:
Doesn't flow already infer this type to the `SourceMapGenerator` class? (defined at https://github.com/babel/babel/blob/7dc5fdb39091b1e9249bcc20487e5cc2aefa44b6/lib/third-party-libs.js.flow#L147)

Reply:
It does, but the `fromObject` [method](https://github.com/babel/babel/blob/7dc5fdb39091b1e9249bcc20487e5cc2aefa44b6/lib/third-party-libs.js.flow#L180) expects a source but we pass it SourceMapGenerator, so maybe I should add `SourceMap | SourceMapGenerator` to that module declaration?

Reply:
I'm thinking something like this `fromObject(obj: SourceMap | SourceMapGenerator): Converter,'. This works without errors.

Reply:
I think that `fromObject(obj: SourceMap)` was correct, and it isn't intended to be called with a `SourceMapGenerator` object.
It should be `.fromObject(map.toJSON())`.

Reply:
Alright. I will make that change.